psydac/api/fem.py

```diff
@@ -730,40 +775,43 @@ def allocate_matrices(self, backend=None):
                     direction = 1 if direction is None else direction
                     flip = [direction]*domain.dim
                     flip[axis] = 1
+
                     if self._func != do_nothing:
-                        global_mats[i,j] = StencilInterfaceMatrix(trial_space, test_space, 
-                                                                  s_d, s_c,
-                                                                  axis, axis,
-                                                                  ext_d, ext_c,
-                                                                  flip=flip)
+                        mat = StencilInterfaceMatrix(trial_fem_space._refined_space[ncells].vector_space,
+                                                     test_fem_space._refined_space[ncells].vector_space,
+                                                     s_d, s_c,
+                                                     axis, axis,
+                                                     ext_d, ext_c,
+                                                     flip=flip)
+                        if not is_conformal:
+                            if all(trn>=tn for trn,tn in zip(trial_fem_space.ncells, test_fem_space.ncells)):
+                                P   = construct_projection_operator(test_fem_space._refined_space[ncells], test_fem_space)
+                                mat = ProductLinearOperator(trial_space, test_space, P, mat)
+                            else:
+                                P   = construct_projection_operator(trial_fem_space, trial_fem_space._refined_space[ncells])
+                                mat = ProductLinearOperator(trial_space, test_space, mat, P)
```

Comment:
Same comment as above, about restriction/prolongation booleans.

Reply:
done in [86f53a3](https://github.com/pyccel/psydac/pull/241/commits/86f53a3e46503da7dac62c58b2edeedc4f631418)